repository/src/main/java/eu/execom/labs/pomodorospringreact/repository/TeamDao.java

```
@@ -1,17 +1,50 @@
 
 import eu.execom.labs.pomodorospringreact.entity.Team;
-import org.springframework.stereotype.Repository;
+import eu.execom.labs.pomodorospringreact.entity.User;
 
+import javax.persistence.NoResultException;
 import javax.persistence.Query;
+import java.util.List;
 
-@Repository
 public class TeamDao extends GenericDao<Team> {
     public Team findByName(String name) {
-        Query query = entityManager.createNativeQuery("SELECT * FROM" +
-                " team WHERE name = ?")
-                .setParameter(1, name);
-        Team team = (Team) query.getSingleResult();
-        return team;
+        try {
+            Query query = entityManager.createNativeQuery("SELECT * FROM" +
+                    " Team WHERE name = ?", Team.class)
+                    .setParameter(1, name);
+            Team team = (Team) query.getSingleResult();
+
+            return team;
+        } catch (NoResultException e) {
+        }
+        return null;
+    }
+
+    public List<User> getTeamMembersByTeamId(Team team) {
+        try {
+            long id = team.getId();
+            Query query = entityManager.createNativeQuery("SELECT A.* FROM User A " +
+                    "LEFT OUTER JOIN UserTeam B ON (A.id = B.userId) " +
+                    "where B.teamId = ?", User.class).setParameter(1, id);
+            List<User> teamMembers = query.getResultList();
+            return teamMembers;
+        } catch (NoResultException e) {
+        }
+        return null;
     }
+
+    public List<Team> getTeamsForUser(User user) {
+        try {
+            long id = user.getId();
+            Query query = entityManager.createNativeQuery("SELECT A.* FROM Team A " +
+                    "LEFT OUTER JOIN UserTeam B ON (A.id = B.teamId) " +
+                    "where B.userId = ?", Team.class).setParameter(1, id);
+            List<Team> teams = query.getResultList();
+            return teams;
+        } catch (NoResultException e) {
+        }
```

This is a bad practice, to "swallow" an exception -> please make a return null from the catch clause.